docs/api/signal.md

```diff
@@ -0,0 +1,217 @@
+# Signal
+
+Signals are just arrays:
+
+```js
+export default []
+```
+
+You populate these arrays with actions:
+
+```js
+export default [
+  someAction,
+  set(state`foo`, 'bar')
+]
+```
+
+You can attach signals at the controller level:
+
+```js
+import {Controller} from 'cerebral'
+import somethingHappened from './signals/somethingHappened'
+
+const Controller({
+  signals: {
+    somethingHappened
+  }
+})
+```
+
+Or you define them with modules:
+
+```js
+import somethingHappened from './signals/somethingHappened'
+
+export default {
+  signals: {
+    somethingHappened
+  }
+}
+```
+
+```js
+import {Controller} from 'cerebral'
+import app from './modules/app'
+
+const Controller({
+  modules: {
+    app
+  }
+})
+```
+
+## Async
+By default signals run completely synchronous, but an action might run asynchronously thus making the signal async. When an action returns a promise it means it runs async.
+
+```js
+function myAction () {
+  return Promise.resolve()
+}
+```
+
+You could also use an **async** function:
+
+```js
+async function myAction () {
+  return {foo: 'bar'}
+}
+```
+
+## Sequence
+Simple format of a sequence is to use an array literal, as explained above:
+
+```js
+import someAction from '../actions/someAction'
+
+export default [
+  someAction
+]
+```
+
+The array is converted to a sequence, but you can also be explicit about it:
+```js
+import {sequence} from 'cerebral'
+import someAction from '../actions/someAction'
+
+export default sequence([
+  someAction
+])
+```
+
+You can name a sequence, which will be displayed in debugger:
+```js
+import {sequence} from 'cerebral'
+import someAction from '../actions/someAction'
+
+export default sequence('my sequence', [
+  someAction
+])
+```
+
+You can compose a sequence into existing sequence. The debugger will show this composition:
+```js
+import someAction from '../actions/someAction'
+import someOtherSequence from './someOtherSequence'
+
+export default [
+  someAction,
+  someOtherSequence
+]
+```
+
+
+## Parallel
+A parallel runs all actions right after each other, even if they return a promise.
```

Comment:
javascript can't run 2 functions at the same time :)
Just compare this sentence with sequence description.
Though I agree with you, current description is very confusing.